sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -523,14 +523,37 @@ class Analyzer(
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       case s @ Sort(ordering, global, p @ Project(projectList, child))
           if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+        val (newOrdering, missing, newChild): (Seq[SortOrder], Seq[Attribute], LogicalPlan) =
+          child match {
+            case Project(
```

Comment:
btw i think this is way too complicated. you can simplify it if you don't specify all the fields, e.g.
```scala
case p1 @ Project(_, w @ Window(_, _, _, _, p2: Project)) =>
```